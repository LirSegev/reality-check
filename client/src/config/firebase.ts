const firebaseConfig = {
	apiKey: 'AIzaSyBcPUPtr_37iqBlCUHpJhZy5cP8FBKqNY4',
	authDomain: 'reality-check-d6fe7.firebaseapp.com',
	databaseURL: 'https://reality-check-d6fe7.firebaseio.com',
	projectId: 'reality-check-d6fe7',
	storageBucket: 'reality-check-d6fe7.appspot.com',
	messagingSenderId: '243239923760',
	appId: '1:243239923760:web:f6bc0b9da0893fa7',
};
export default firebaseConfig;
