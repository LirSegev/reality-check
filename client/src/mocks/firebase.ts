import { mockFirebase } from 'ts-mock-firebase';

const firebase = mockFirebase();

export default firebase;

export const app = firebase.initializeApp({});
