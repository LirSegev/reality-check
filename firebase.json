{
	"functions": {
		"predeploy": [
			"npm --prefix \"$RESOURCE_DIR\" run lint",
			"npm --prefix \"$RESOURCE_DIR\" run build"
		],
		"source": "functions"
	},
	"hosting": {
		"site": "realitycheck",
		"public": "client/build",
		"ignore": ["firebase.json", "**/.*", "**/node_modules/**"],
		"rewrites": [
			{
				"source": "**",
				"destination": "/index.html"
			}
		],
		"predeploy": ["npm --prefix ./client run build"]
	},
	"firestore": {
		"rules": "firestore.rules",
		"indexes": "firestore.indexes.json"
	}
}
