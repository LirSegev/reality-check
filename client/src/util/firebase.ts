import * as firebase from 'firebase/app';
import { getCurrentPlayer } from './db';

export function signOut(isAdmin?: boolean) {
	if (!isAdmin) {
		const gameId = localStorage.getItem('gameId');
		if (gameId)
			getCurrentPlayer().then(player => {
				if (player && player.messagingToken) {
					const removeDeviceFromDeviceGroup = firebase
						.functions()
						.httpsCallable('removeDeviceFromDeviceGroup');

					removeDeviceFromDeviceGroup({
						token: player.messagingToken,
						gameId,
					});
				} else {
					console.error(new Error("Player doesn't have messagingToken"));
				}
			});
		else console.error(new Error('Error no gameID'));
	}
	firebase.auth().signOut();
}
