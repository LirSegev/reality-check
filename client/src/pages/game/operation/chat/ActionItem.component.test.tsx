import { cleanup, render } from '@testing-library/react';
import * as firebase from 'firebase';
import React from 'react';
import { Card } from 'react-onsenui';

import ActionItemImport from './ActionItem.component';

let ActionItem: typeof ActionItemImport;

beforeAll(() => {
	jest.mock('react-onsenui', () => ({
		Card: (
			props: React.PropsWithChildren<React.ComponentProps<typeof Card>>
		) => <div className={props.className}>{props.children}</div>,
	}));

	ActionItem = require('./ActionItem.component').default;
});
afterEach(cleanup);

it('renders', () => {
	const { asFragment } = render(
		<ActionItem
			action={{
				subject: {
					uid: 'someUid',
					displayName: 'someName',
				},
				action: 'questioned',
				timestamp: new firebase.firestore.Timestamp(34, 0),
			}}
		/>
	);

	expect(asFragment()).toMatchSnapshot();
});

it('displays player name and action', () => {
	const { getByText } = render(
		<ActionItem
			action={{
				subject: {
					uid: 'someUid',
					displayName: 'someName',
				},
				action: 'questioned',
				timestamp: new firebase.firestore.Timestamp(34, 0),
			}}
		/>
	);

	expect(getByText('someName', { exact: false })).toBeVisible();
	expect(getByText('questioned', { exact: false })).toBeVisible();
});

it.each([
	[
		{
			subject: {
				uid: 'someUid',
				displayName: 'someName',
			},
			action: 'questioned',
			timestamp: new firebase.firestore.Timestamp(10, 0),
		},
		'someName questioned a suspect',
	],
	[
		{
			subject: {
				uid: 'anotherUid',
				displayName: 'anotherName',
			},
			action: 'marked',
			object: {
				id: 1,
				name: 'Tomas',
			},
			timestamp: new firebase.firestore.Timestamp(10, 0),
		},
		'anotherName marked Tomas as the main suspect',
	],
	[
		{
			subject: {
				uid: 'someUid',
				displayName: 'someName',
			},
			action: 'collected',
			object: {
				type: 'intelligence',
			},
			timestamp: new firebase.firestore.Timestamp(10, 0),
		},
		'someName collected a point',
	],
] as [React.ComponentProps<typeof ActionItem>['action'], string][])(
	'displays correct message for each action type',
	(action, expected) => {
		const { getByText } = render(<ActionItem action={action} />);

		expect(getByText(expected)).toBeVisible();
	}
);
