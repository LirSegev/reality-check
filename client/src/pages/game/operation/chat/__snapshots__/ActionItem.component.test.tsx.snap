// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    style="clear: both;"
  />
  <div
    class="chatItem middle action"
  >
    someName questioned a suspect
  </div>
</DocumentFragment>
`;
